tests/test_spl_build.py

```diff
@@ -67,3 +71,33 @@ def test_execute_with_additional_args(mock_execute: MagicMock, spl_build: SplBui
 
     # Assertions
     mock_execute.assert_called_once_with(["build.bat", "-buildKit", "defaultKit", "-variants", "my_var", "-target", "all", "-reconfigure", "-j", "4"])
+
+
+def test_create_artifacts_archive(spl_build: SplBuild) -> None:
+    """
+    Test that additional arguments are passed to the command line executor correctly.
+    """
+    # Setup mock
+    test_dir = TestUtils.create_clean_test_dir(str(spl_build.build_dir.absolute()))
+    test_dir_2 = TestUtils.create_clean_test_dir("outside_build")
+    outside_file = test_dir_2.write_file("outside_file")
+    assert test_dir.path.exists()
+    file_1 = test_dir.write_file(str(test_dir.path.joinpath("out", "bla.exe")))
+    file_2 = test_dir.write_file(str(test_dir.path.joinpath("blub.exe")))
+    test_folder = TestUtils.create_clean_test_dir(str(Path.joinpath(test_dir.path, "some_folder")))
+    test_dir.write_file(str(test_folder.path.joinpath("blub.exe")))
+    archive_dir = spl_build.create_artifacts_archive([test_folder.path, file_1, file_2, outside_file])
+    archive_json = spl_build.create_artifacts_json([test_folder.path, file_1, file_2, outside_file])
+    assert archive_dir.exists()
+    with zipfile.ZipFile(archive_dir) as zip_ref:
+        file_list = zip_ref.namelist()
+        assert file_list == ["some_folder/blub.exe", f"out/{file_1.name}", file_2.name, outside_file.name]
+    assert archive_json.exists()
+    assert dict(json.loads(archive_json.read_text())) == {'variant': 'my_var',
+                                                          'build_kit': 'defaultKit',
+                                                          'artifacts': [
+                                                              "some_folder/blub.exe",
+                                                              f"out/{file_1.name}",
+                                                              file_2.name,
+                                                              outside_file.name]
+                                                          }

```

Comment:
This test is not readable. I can not understand it. Try to split it in somehow in three parts: inputs, run, assertions

Reply:
Updated tests